src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -198,22 +357,36 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submiss
 
         <Footer>
           <Flex justifyContent="justify-center" spaceX="space-x-3">
+            <Button
+              size="xs"
+              icon={() => <BpIcon icon={IconNames.DOUBLE_CHEVRON_LEFT} />}
+              variant="light"
+              onClick={() => resetPage()}
+              disabled={page <= 0}
+            />
             <Button
               size="xs"
               icon={() => <BpIcon icon={IconNames.ARROW_LEFT} />}
               variant="light"
-              onClick={() => table.previousPage()}
-              disabled={!table.getCanPreviousPage()}
+              onClick={() => setPage(page - 1)}
+              disabled={page <= 0}
             />
             <Bold>
-              Page {table.getState().pagination.pageIndex + 1} of {table.getPageCount()}
+              Page {page + 1} of {Math.ceil(totalRows / pageSize)}
             </Bold>
             <Button
               size="xs"
               icon={() => <BpIcon icon={IconNames.ARROW_RIGHT} />}
               variant="light"
-              onClick={() => table.nextPage()}
-              disabled={!table.getCanNextPage()}
+              onClick={() => setPage(page + 1)}
+              disabled={page >= Math.ceil(totalRows / pageSize) - 1}
+            />
+            <Button
+              size="xs"
+              icon={() => <BpIcon icon={IconNames.DOUBLE_CHEVRON_RIGHT} />}
+              variant="light"
+              onClick={() => setPage(Math.ceil(totalRows / pageSize) - 1)}
+              disabled={page >= Math.ceil(totalRows / pageSize) - 1}
```

Comment:
There's quite a lot of duplication here. Can we assign the max page number to a `const` instead?

Reply:
implemented!